modules/xfem/src/base/XFEM.C

```diff
@@ -1430,6 +1430,11 @@ XFEM::cutMeshWithEFA(NonlinearSystemBase & nl, AuxiliarySystem & aux)
     }
 
     elem_to_delete->nullify_neighbors();
+
+    // remove the property storage of deleted element/side
+    (*_material_data)[0]->eraseProperty(elem_to_delete);
+    (*_bnd_material_data)[0]->eraseProperty(elem_to_delete);
+
```

Comment:
It looks like there are several places in the XFEM class/file that you delete elements.  And you remove boundary info, erase material properties, etc.  along with the actual element deletions.  Would it be better to have a special removeElem function that safely did all these things atomically? - That way you won't have to worry about forgetting to do any of these things whenever you are deleting elements.  Or are the situations for element deletion different enough that they all need different subsets of things done?

Reply:
You are correct that there are other three places that call element deletion. However those deletion is for elements on the displaced mesh, and there is no need to remove their material properties because undisplaced and displaced mesh share same material properties.